contracts/PushComm/PushCommV3.sol

```diff
@@ -471,4 +474,63 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3 {
         userToChannelNotifs[msg.sender][_channel] = notifSetting;
         emit UserNotifcationSettingsAdded(_channel, msg.sender, _notifID, notifSetting);
     }
+
+    ///@notice Wallet PGP attach code starts here
+
+    function setFeeAmount(uint256 _feeAmount) external onlyPushChannelAdmin {
+        FEE_AMOUNT = _feeAmount;
+    }
+
+    function registerUserPGP(bytes calldata _data, string calldata _pgp, bool _isNFT) external {
+        uint256 fee = FEE_AMOUNT;
+        PROTOCOL_POOL_FEE += fee;
+        IERC20(PUSH_TOKEN_ADDRESS).safeTransferFrom(msg.sender, address(this), fee);
+        bytes32 hash = keccak256(_data);
+        if (!_isNFT) {
+            (, address _wallet) = abi.decode(_data, (string, address));
```

Comment:
The CAIP format for an address looks like this:

`eip155:11155111:0x99A08ac6254dcf7ccc37CeC662aeba8eFA666666`

where chainId (11155111) is also passed. 

However, while decoding we only consider (string, address ) instead of (string, uint256, address)

If this isn't intended, it needs to be fixed.